File: domain/usecases/get_featured_packages.py
from typing import List
from domain.entities import Package
from domain.repositories import FeaturedRepository

class GetFeaturedPackagesUseCase:
    def __init__(self, featured_repo: FeaturedRepository):
        self.featured_repo = featured_repo

    def execute(self) -> List[Package]:
        return self.featured_repo.get_featured_packages()

File: src-rust/src/domain/usecases/get_featured_packages.rs
use std::sync::Arc;
use crate::domain::{entities::Package, repositories::FeaturedRepository};

pub struct GetFeaturedPackagesUseCase {
    pub appstream_repo: Arc<dyn FeaturedRepository>,
}
impl GetFeaturedPackagesUseCase {
    pub fn new(appstream_repo: Arc<dyn FeaturedRepository>) -> Self {
        Self { appstream_repo }
    }
    pub async fn execute(&self) -> Vec<Package> {
        self.appstream_repo.get_featured_packages().await
    }
}
